File: yandex_autumn_intern_day_2023/2. Сложить и вычесть/solution_.py
import sys
from typing import List

def calculate(s: str) -> int:
    result: int = 0
    current_number: int = 0
    operation: str = '+'
    l: int = len(s) - 1
    for i in range(len(s)):
        c: str = s[i]

        if c.isdigit():
            current_number = current_number * 10 + int(c)

        if not c.isdigit() or i == l:
            if operation == '+':
                result += current_number
            elif operation == '-':
                result -= current_number
            current_number = 0
            operation = c 

    return result

def main():
    s: str = input()
    print(calculate(s))

if __name__ == '__main__':
    main()

File: yandex_autumn_intern_day_2023/2. Сложить и вычесть/solution.cs
using System;

class Program 
{
    static void Main() 
    {
        string s = Console.ReadLine();
        Console.WriteLine(Calculate(s));
    }

    private static long Calculate(string s)
    {
        long result = 0;
        long currentNumber = 0;
        char operation = '+';
        int len = s.Length - 1;

        for (int i = 0; i < s.Length; i++)
        {
            char c = s[i];

            if (char.IsDigit(c))
                currentNumber = currentNumber * 10 + (c - '0');

            if (!char.IsDigit(c) || i == len)
            {
                if (operation == '+')
                {
                    result += currentNumber;
                }
                else if (operation == '-')
                {
                    result -= currentNumber;
                }
                
                currentNumber = 0;
                operation = c;
            }
        }
        return result;
    }
}